src/mono/mono/mini/local-propagation.c

```diff
@@ -350,7 +350,7 @@ mono_strength_reduction_ins (MonoCompile *cfg, MonoInst *ins, const char **spec)
 			ins->opcode = OP_INEG;
 		} else if ((ins->opcode == OP_LMUL_IMM) && (ins->inst_imm == -1)) {
 			ins->opcode = OP_LNEG;
-		} else if (ins->inst_imm > 0) {
+		} else if (ins->inst_imm > 0 && ins->inst_imm <= UINT32_MAX) {
 			int power2 = mono_is_power_of_two (GTMREG_TO_UINT32 (ins->inst_imm));
```

Comment:
Why doesn't this fail on other platforms ?

Reply:
This only fails in debug mode. Not sure if other platforms are checking in debug mode or not. But power arch has 8 byte registers and many test cases fails due to this wherever Double.TryFormat method is used.

Reply:
@vargaz 
After figuring out a lot on power, identified that the problem is multiplication instruction is getting changed to left shift instruction due to the data loss which is happening for one of the 64 bit values as mono_is_power_of_two accepts uint32 and not uint64. Hence added a safety check here to avoid the data loss. There is no other place where this issue can be fixed.